File: Lista-15/go/5.go
package main
import "fmt"

func FazerSom(animais []Animal) {
    for _, animal := range animais {
        fmt.Println(animal.Som())
    }
}

func main() {
    animais := []Animal{Cachorro{}, Gato{}}
    FazerSom(animais)
}

File: Lista-15/java/5.java
public class Main {
    public static void fazerSom(Animal[] animais) {
        for (Animal animal : animais) {
            System.out.println(animal.som());
        }
    }

    public static void main(String[] args) {
        Animal[] animais = {new Cachorro(), new Gato()};
        fazerSom(animais);
    }
}
